src/mlpack/methods/ann/augmented/tasks/copy_impl.hpp

```diff
@@ -0,0 +1,66 @@
+/**
+ * @file copy_impl.hpp
+ * @author Konstantin Sidorov
+ *
+ * Implementation of CopyTask class
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_AUGMENTED_TASKS_COPY_IMPL_HPP
+#define MLPACK_METHODS_AUGMENTED_TASKS_COPY_IMPL_HPP
+
+// In case it hasn't been included yet.
+#include "copy.hpp"
+
+using mlpack::math::RandInt;
+
+namespace mlpack {
+namespace ann /* Artificial Neural Network */ {
+namespace augmented /* Augmented neural network */ {
+namespace tasks /* Task utilities for augmented */ {
+
+CopyTask::CopyTask(const size_t maxLength, const size_t nRepeats) :
+    maxLength(maxLength),
+    nRepeats(nRepeats)
+{
+  assert(maxLength > 1);
+  assert(nRepeats > 0);
+  // Just storing task-specific parameters.
+}
+
+void CopyTask::Generate(arma::field<arma::mat>& input,
+                        arma::field<arma::mat>& labels,
+                        const size_t batchSize)
+{
+  input = arma::field<arma::mat>(batchSize);
+  labels = arma::field<arma::mat>(batchSize);
+  for (size_t i = 0; i < batchSize; ++i) {
+    // Random uniform length from [2..maxLength]
+    size_t size = RandInt(2, maxLength+1);
+    arma::colvec vecInput = arma::randi<arma::colvec>(
+      size, arma::distr_param(0, 1));
+    arma::colvec vecLabel = arma::conv_to<arma::colvec>::from(
+      arma::repmat(vecInput, nRepeats, 1));
+    size_t totSize = vecInput.n_elem + vecLabel.n_elem;
+    input(i) = arma::zeros(totSize, 2);
```

Comment:
I think it's a good idea to add some more comments here, and maybe a new line between each sub task (input generation, labels generation), to improve the readability.